src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java

```diff
@@ -52,54 +51,58 @@
     /** Previous sibling. */
     private DetailAstImpl previousSibling;
 
+    /** First child of this DetailAST. */
+    private DetailAstImpl firstChild;
+
+    /** First sibling of this DetailAST.*/
+    private DetailAstImpl nextSibling;
+
+    /** Text of this DetailAST. */
+    private String text;
+
+    /** The type of this DetailAST. */
+    private int type;
+
+    /**
+     * All tokens on COMMENTS channel to the left of the current token up to the
+     * preceding token on the DEFAULT_TOKEN_CHANNEL.
+     */
+    private List<Token> hiddenBefore;
+
+    /**
+     * All tokens on COMMENTS channel to the right of the current token up to the
+     * next token on the DEFAULT_TOKEN_CHANNEL.
+     */
+    private List<Token> hiddenAfter;
+
     /**
      * All token types in this branch.
      * Token 'x' (where x is an int) is in this branch
      * if branchTokenTypes.get(x) is true.
      */
     private BitSet branchTokenTypes;
 
-    @Override
-    public void initialize(Token token) {
-        super.initialize(token);
-        lineNo = token.getLine();
-
-        // expect columns to start @ 0
-        columnNo = token.getColumn() - 1;
-    }
-
-    @Override
-    public void initialize(AST ast) {
-        final DetailAstImpl detailAst = (DetailAstImpl) ast;
-        setText(detailAst.getText());
-        setType(detailAst.getType());
-        lineNo = detailAst.getLineNo();
-        columnNo = detailAst.getColumnNo();
-        hiddenAfter = detailAst.getHiddenAfter();
-        hiddenBefore = detailAst.getHiddenBefore();
-    }
-
-    @Override
-    public void setFirstChild(AST ast) {
-        clearBranchTokenTypes();
-        clearChildCountCache(this);
-        super.setFirstChild(ast);
-        if (ast != null) {
-            ((DetailAstImpl) ast).setParent(this);
-        }
+    /**
+     * Initializes this DetailAstImpl.
+     *
+     * @param tokenType the type of this DetailAstImpl
+     * @param tokenText the text of this DetailAstImpl
+     */
+    public void initialize(int tokenType, String tokenText) {
```

Comment:
can we make it as constructor?

Reply:
We can, and I considered this. I ultimately decided not to create a constructor in order to minimize changes; `DetailAstImpl` already had two `initialize` methods that were overrides from the `AST` interface, I just modified this one to eliminate the extra call to `setText` after initialization. I also decided not to add a constructor to minimize complexity in `DetailAstImpl`; if I add a constructor, I think that I should remove the `initialize` methods. What do you think?

Reply:
Constructor is supposed to be block of code to initialize object. 
I do not think anyone will call this initialize method more than one time, so I do not see a reason for it. Other approach is to make static factory methods for object creation.

I do not like current logic with init methods as well, I would suggest to rewrite them (in other task)

Reply:
> I do not like current logic with init methods as well, I would suggest to rewrite them

Can you give more details?

>  (in other task)

In another PR/ issue?

Reply:
Issue opened at https://github.com/checkstyle/checkstyle/issues/10557 for further discussion. Thanks for bringing this up.